actors/sunbird-lms-mw/actors/common/src/main/java/org/sunbird/learner/actors/qrcodedownload/QRCodeDownloadManagementActor.java

```diff
@@ -0,0 +1,223 @@
+import static org.sunbird.common.models.util.ProjectUtil.getConfigValue;
+
+/**
+ * @Author : Rhea Fernandes
+ * This actor is used to create an html file for all the qr code images that are
+ * linked to courses that are created userIds given
+ */
+@ActorConfig(
+        tasks = {"downloadQRCodes"},
+        asyncTasks = {}
+)
+public class QRCodeDownloadManagementActor extends BaseActor {
+    private static final List<String> fields = Arrays.asList("identifier", "dialcodes", "name");
+    private static final Map<String, String> filtersHelperMap = new HashMap<String, String>() {{
+        put(JsonKey.USER_IDs, JsonKey.CREATED_BY);
+        put(JsonKey.STATUS, JsonKey.STATUS);
+        put(JsonKey.CONTENT_TYPE, JsonKey.CONTENT_TYPE);
+    }};
+    private static Util.DbInfo courseDialCodeInfo =
+            Util.dbInfoMap.get(JsonKey.SUNBIRD_COURSE_DIALCODES_DB);
+    private static CassandraOperation cassandraOperation = ServiceFactory.getInstance();
+
+    @Override
+    public void onReceive(Request request) throws Throwable {
+        Util.initializeContext(request, TelemetryEnvKey.QR_CODE_DOWNLOAD);
+        ExecutionContext.setRequestId(request.getRequestId());
+        String requestedOperation = request.getOperation();
+        switch (requestedOperation) {
+            case "downloadQRCodes":
+                downloadQRCodes(request);
+                break;
+
+            default:
+                onReceiveUnsupportedOperation(requestedOperation);
+                break;
+        }
+    }
+
+    /**
+     * The request must contain list of userIds (Users Ids of people who have created courses)
+     * @param request
+     */
+    private void downloadQRCodes(Request request) {
+        Map<String, String> headers = (Map<String, String>) request.getRequest().get(JsonKey.HEADER);
+        Map<String, Object> requestMap = (Map<String, Object>) request.getRequest().get(JsonKey.FILTER);
+        requestMap.put(JsonKey.CONTENT_TYPE, "course");
+        Map<String, Object> searchResponse = searchCourses(requestMap, headers);
+        List<Map<String, Object>> contents = (List<Map<String, Object>>) searchResponse.get("contents");
+        if (CollectionUtils.isEmpty(contents))
+            throw new ProjectCommonException(
+                    ResponseCode.errorUserHasNotCreatedAnyCourse.getErrorCode(),
+                    ResponseCode.errorUserHasNotCreatedAnyCourse.getErrorMessage(),
+                    ResponseCode.CLIENT_ERROR.getResponseCode());
+        Map<String, List<String>> dialCodesMap = contents.stream()
+                .filter(content -> content.get("dialcodes") != null)
+                .filter(content -> content.get("name") != null)
+                .collect(Collectors.toMap(content -> ((String) content.get("identifier")) + "<<<" + (String) content.get("name"), content -> (List) content.get("dialcodes")));
+        File file = generateCSVFile(dialCodesMap);
```

Comment:
what if the dialcodesMap is empty, it should not throw error while processing the file. It should throw invalid course as client error

Reply:
Done